### PR TITLE
[ADAM-625] Enable globbing for BAM.

### LINKED ISSUE #625: Cannot glob with loadBam
Currently, `loadBam` reads the header from a BAM file before loading it. The way this code is written, you can't provide a glob containing multiple BAM files.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #625. Elaborates the glob before loading the header.

### CODE REVIEW

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala [hunk 1]
```diff
@@ -85,6 +85,16 @@ class TransformArgs extends Args4jBase with ADAMSaveAnyArgs with ParquetArgs {
...
   @Args4jOption(required = false, name = "-sort_fastq_output", usage = "Sets whether to sort the FASTQ output, if saving as FASTQ. False by default. Ignored if not saving as FASTQ.")
   var sortFastqOutput: Boolean = false
+  @Args4jOption(required = false, name = "-force_load_bam", usage = "Forces Transform to load from BAM.")
+  var forceLoadBam: Boolean = false
+  @Args4jOption(required = false, name = "-force_load_sam", usage = "Forces Transform to load from SAM.")
+  var forceLoadSam: Boolean = false
+  @Args4jOption(required = false, name = "-force_load_fastq", usage = "Forces Transform to load from unpaired FASTQ.")
+  var forceLoadFastq: Boolean = false
+  @Args4jOption(required = false, name = "-force_load_fastq", usage = "Forces Transform to load from interleaved FASTQ.")
+  var forceLoadIFastq: Boolean = false
```
Comment (laserson) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala]: did you mean fasta? or just one extra cmd+p?

**Comment by laserson** [comment]
lgtm after the build is fixed

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala (hunk 1 above)
Comment (fnothaft) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala]: There was a missing `i --> interleaved`; I had fixed it locally but forgot to commit before pushing.

**Comment by massie** [comment]
Thanks, Frank!

**Pull request merged**